### ISSUE TITLE
How to install linux-headers？

### ISSUE THREAD

**Issue by xiayang0521** [ophub/amlogic-s9xxx-armbian#126]
When I use : 
apt install linux-headers,
there is no one matches the compiled linux kernel.
So how to install linux-headers of the compiled linux kernel or genrate a linux-headers-**.deb file of the compiled linux kernel?

**Comment by xiayang0521** [comment]
But I failed to install zfs module:
apt install zfs-dkms 
without installing the linux-headers-*.deb.

> https://github.com/ophub/amlogic-s9xxx-armbian#update-armbian-kernel
> 
> The headers files in the kernel is installed in the /use/local/include directory. When compiling the application, add -I /usr/local/include to the CFLAG parameter of GCC to find the headers files.

 But I failed to install zfs module:
apt install zfs-dkms 
without installing the linux-headers-*.deb.

**Comment by ophub** [comment]
```
sudo apt install dkms
sudo apt install zfs-dkms —no-install-recommends
sudo modprobe zfs
sudo apt install zfsutils-linux
```

**Comment by xiayang0521** [comment]
Hi, Ophub Great, thank you very much for your suggestions.After running the following cmds:sudo apt install dkms sudo apt install zfs-dkms —no-install-recommendssudo modprobe zfsthe error shows:
modprobe: FATAL: Module zfs not found in directory /lib/modules/5.4.181-flippy-70+o




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2022年2月28日(星期一) 晚上10:14
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [ophub/amlogic-s9xxx-armbian] How to install linux-headers？ (Issue #126)





 sudo apt install dkms sudo apt install zfs-dkms —no-install-recommends sudo modprobe zfs sudo apt install zfsutils-linux  
&mdash;
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by ophub** [comment]
Continue to execute the fourth command to install

**Comment by xiayang0521** [comment]
After the fourth command is&nbsp;executed, then&nbsp;running:
modprobe zfs&nbsp;
keeps show the same error.


the following msgs show that the apt-install-pkgs have been all installed but zfs module is not installed:
***@***.***:/mnt# sudo apt install dkms
Reading package lists... Done
Building dependency tree
Reading state information... Done
dkms is already the newest version (2.8.1-5ubuntu2).
0 upgraded, 0 newly installed, 0 to remove and 28 not upgraded.
***@***.***:/mnt# sudo apt install zfs-dkms --no-install-recommends
Reading package lists... Done
Building dependency tree
Reading state information... Done
zfs-dkms is already the newest version (2.1.2-0york0~20.04).
0 upgraded, 0 newly installed, 0 to remove and 28 not upgraded.
***@***.***:/mnt#&nbsp; apt install zfsutils-linux
Reading package lists... Done
Building dependency tree
Reading state information... Done
zfsutils-linux is already the newest version (2.1.2-0york0~20.04).
0 upgraded, 0 newly installed, 0 to remove and 28 not upgraded.
***@***.***:/mnt# modprobe zfs
modprobe: FATAL: Module zfs not found in directory /lib/modules/5.4.181-flippy-70+o



Actually, in some orignal build armbian systems, after running:
apt install linux-headers-current-meson64&nbsp;
then run the cmds of your suggestion, the zfs module can be successfully installed.






------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2022年3月1日(星期二) 晚上7:53
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [ophub/amlogic-s9xxx-armbian] How to install linux-headers？ (Issue #126)





 
Continue to execute the fourth command to install
 
&mdash;
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by petesimon** [comment]
>sudo apt install dkms -y
> sudo apt install zfs-dkms --no-install-recommends -y
> sudo modprobe zfs
> sudo apt install zfsutils-linux -y

Thank  you. I have (old) ubuntu jammy 22.04 from the "releases" page installed on an [amlogic s905x3 tv box](https://archive.org/details/x88-pro-x3-s905x3-tvbox-android-firmware-backup) ( [YT video](https://youtu.be/znSER4wrs2Y) ). It's working fine. The USB 3.0 and Gigabit Ethernet ports offer a decent solution to make a NAS with a USB 3.0 external hard drive. I am currently using an ext4 filesystem but I wanted to try ZFS. So, I ran the commands above and they worked. Then I ran the commands below to get some information.

`lsmod | grep zfs`
`sudo zfs version`
`sudo zpool status`
`man zfs`
`man zpool`

and they work fine. Now I must research heaps of information about ZFS and how to use it, hehe. Thanks again. ☺♥
